File: 2010/q/a/store_credit.pl
$N = <>;
while($n++<$N) {
	$C = <> + 0;
	$I = <> + 0;
	@P = split(' ',<>);
	$done = false;
	for($x = 0; $x < $I && $done == false; $x++) {
		for($y = ($x+1); $y < $I; $y++) {
			if(@P[$x] + @P[$y] == $C) {
				$done = true;
				print("Case #$n: " . ($x+1) . " " . ($y+1) . "\n");
				break;
			}
		}		
	}
}

File: 2010/q/a/store_credit.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
	int N, c=0;
	cin >> N;
	while(c++ < N) {
		int I,C;
		cin >> C >> I;
		vector<int> vP;
		while(I-->0) {
			int P;
			cin >> P;
			vP.push_back(P);
		}
		bool done = false;
		for(int x = 0; !done && x < vP.size(); x++) {
			for(int y = x+1; y < vP.size(); y++) {
				if(vP[x] + vP[y] == C) {
					cout << "CASE #" << c << ": " << x+1 << " " << y+1 << endl;
					done = true;
					break;
				}
			}
		}
	}
}
